### DISCUSSION TITLE
Purpose of `crsql_wdbreplicator`

### DISCUSSION THREAD

**Discussion by AlexErrant** [vlcn-io/cr-sqlite#103] [Q&A] [answered]
I'm trying to build a RESTish client/server layer. I'm basing it heavily off of the [WholeDbReplicator](https://github.com/vlcn-io/cr-sqlite/blob/main/js/sync-reference/whole-db/src/WholeDbReplicator.ts). Is the entire purpose of [`crsql_wdbreplicator`](https://github.com/vlcn-io/cr-sqlite/blob/8f835a836685b8576d2f18ceb22e09687368ef95/js/sync-reference/whole-db/src/WholeDbReplicator.ts#L112) to alert remote nodes that a change has occurred in the local sqlite? The reason I ask is because I want to delete (from my code) [`installTriggers`](https://github.com/vlcn-io/cr-sqlite/blob/8f835a836685b8576d2f18ceb22e09687368ef95/js/sync-reference/whole-db/src/WholeDbReplicator.ts#L144-L174). If the answer to my question is "yes", I would feel safe deleting `installTriggers` since live updates aren't important in my app's planned database sync lifecycle.

**Comment by tantaman** [comment, marked as answer]
> Is the entire purpose of [crsql_wdbreplicator](https://github.com/vlcn-io/cr-sqlite/blob/8f835a836685b8576d2f18ceb22e09687368ef95/js/sync-reference/whole-db/src/WholeDbReplicator.ts#L112) to alert remote nodes that a change has occurred in the local sqlite?

Yep, that's right. Completely safe to delete `installTriggers` if you don't need to be alerted to changes in the database

You might also be interested in the client-server sync reference implementation I've been working on
https://vlcn.io/docs/replicators as another example of a sync implementation.

The main difference with these is that they take more of a "streaming approach" rather than the simple "poke" approach.

**Reply by AlexErrant** [reply]
Wow, I didn't know that Sqlite could call external code via triggers, and upon failure roll back a transaction. That's powerful as heck.

**Reply by tantaman** [reply]
In the latest version I’m switching from triggers to the “update_hook”
which is called post rather than pre commit.

**Reply by tantaman** [reply]
https://github.com/vlcn-io/cr-sqlite/blob/8f835a836685b8576d2f18ceb22e09687368ef95/js/rx-tbl/src/tblrx.ts#L31-L37

**Reply by AlexErrant** [reply]
Ah yeah, a post-commit hook makes more sense for notification/reactivity semantics.
